src/provider/model/Storage.ts

```diff
@@ -1,13 +1,95 @@
 import localforage from 'localforage';
+import {injectable} from 'inversify';
 
-export const settingsStorage = localforage.createInstance({
-    driver: localforage.INDEXEDDB,
-    name: 'notifications',
-    storeName: 'settings'
-});
-
-export const notificationStorage = localforage.createInstance({
-    driver: localforage.INDEXEDDB,
-    name: 'notifications',
-    storeName: 'notifications'
-});
+import {AsyncInit} from '../controller/AsyncInit';
+
+/**
+ * List of available storages.
+ */
+export const enum StorageMap {
+    NOTIFICATIONS = 'notifications',
+    SETTINGS = 'settings'
+}
+
+@injectable()
+export class Storage extends AsyncInit {
+    /**
+     * This value should be updated any time changes are made to the notification shape.
+     * Must remain in number format.  LocalForage does not support semver.
+     */
+    public static readonly DATABASE_VERSION: number = 1.0;
+    private _storages: Map<string, LocalForage>;
+
+    constructor() {
+        super();
+        this._storages = new Map<string, LocalForage>();
+
+        [StorageMap.NOTIFICATIONS, StorageMap.SETTINGS].forEach(storeName => {
+            this.add({
+                driver: localforage.INDEXEDDB,
+                version: Storage.DATABASE_VERSION,
+                name: 'notifications',
+                storeName
+            });
+        });
+    }
+
+    protected async init(): Promise<void> {
+        await Promise.all([...this._storages.values()].map(storage => storage.ready()));
+
+        const settingsStorage = this.get(StorageMap.SETTINGS);
+
+        const storedVersion = await settingsStorage.getItem<number>('dbVersion');
+        const dbVersion = settingsStorage.config().version!;
```

Comment:
For my understanding - will this always equal `DATABASE_VERSION`?

Reply:
You would be correct

Reply:
Fixed it to just reference `DATABASE_VERSION`

Reply:
Cool. Same change on line 67 for consistency?